mitodl/open-discussions
# Simplify the docker/compose setup

#### Pre-Flight checklist


#### What are the relevant tickets?
N/A

#### What's this PR do?
I'd like to try out some features in newer elasticsearch/opensearch, but before I do that I'd like to have a cleaner foundation to work on with docker - this cleans up the configuration to be in line with how mitx online is setup by using compose profiles and multistage docker builds.

#### How should this be manually tested?
You should be able to run `docker-compose up` as normal and not have the notebook container start. Running `docker-compose --profile notebook up` should give you the notebook.
